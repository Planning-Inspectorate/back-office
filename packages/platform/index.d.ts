import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
