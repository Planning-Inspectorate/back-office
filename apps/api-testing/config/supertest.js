// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
