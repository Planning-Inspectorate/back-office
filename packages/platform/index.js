export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
