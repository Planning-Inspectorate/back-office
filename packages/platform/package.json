{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
