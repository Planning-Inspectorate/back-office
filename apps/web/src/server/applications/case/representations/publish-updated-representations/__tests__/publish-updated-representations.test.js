import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../../../testing/index.js';
import { publishableRepresentationsFixture } from '../../__fixtures__/publishable-representations.fixture.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);

const baseUrl =
	'/applications-service/case/1/relevant-representations/select-representations-for-publishing';

const mockCaseData = {
	title: 'mock title'
};

describe('publish-updated-representations.controller', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	afterAll(() => {
		nock.cleanAll();
	});

	describe('GET /applications-service/:caseId/relevant-representations/select-representations-for-publishing', () => {
		const nocks = () => {
			nock('http://test/').get('/applications/1').reply(200, mockCaseData);
			nock('http://test/')
				.get('/applications/1/representations/publishable')
				.reply(200, publishableRepresentationsFixture);
		};
		nocks();

		it('should render the page', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain('Select representations for publishing');
			expect(element.innerHTML).toContain('The queue contains <strong>3</strong> representations.');
		});
	});
});
