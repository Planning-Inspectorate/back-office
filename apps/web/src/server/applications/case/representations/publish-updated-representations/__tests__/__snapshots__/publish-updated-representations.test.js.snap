// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`publish-updated-representations.controller GET /applications-service/:caseId/relevant-representations/select-representations-for-publishing should render the page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\"><span class=\\"govuk-caption-m\\"> Publishing queue</span>
    <h2 class=\\"govuk-heading-l\\"> Select representations for publishing</h2>
    <p class=\\"govuk-body\\"><strong>Project name:</strong> mock title</p>
    <p class=\\"govuk-body\\">The queue contains <strong>3</strong> representations.</p>
    <br>
    <p class=\\"govuk-body\\">You must select the representations to publish</p>
    <p class=\\"govuk-body\\"><strong><span id=\\"number-of-selected-representations\\">0</span> representations on this page are selected.</strong>
    </p>
    <form method=\\"post\\">
        <table class=\\"govuk-table pins-files-list__table pins-table\\">
            <thead class=\\"govuk-table__head\\">
                <tr class=\\"govuk-table__row\\">
                    <th scope=\\"col\\" class=\\"govuk-table__header white-space--nowrap\\">
                        <div class=\\"govuk-!-display-none\\" id=\\"select-all-representations-container\\">
                            <div class=\\"govuk-form-group\\">
                                <div class=\\"govuk-checkboxes\\" data-module=\\"govuk-checkboxes\\">
                                    <div class=\\"govuk-checkboxes__item\\">
                                        <input class=\\"govuk-checkboxes__input\\" id=\\"representationId\\" name=\\"select-all-representations\\"
                                        type=\\"checkbox\\" value=\\"\\">
                                        <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"representationId\\">Select all representations</label>
                                    </div>
                                </div>
                            </div>
                        </div><span class=\\"govuk-!-display-inline-block\\">Select</span>
                    </th>
                    <th scope=\\"col\\" class=\\"govuk-table__header\\">Reference</th>
                    <th scope=\\"col\\" class=\\"govuk-table__header\\">From</th>
                    <th scope=\\"col\\" class=\\"govuk-table__header\\">Status</th>
                </tr>
            </thead>
            <tbody class=\\"govuk-table__body\\">
                <tr class=\\"govuk-table__row\\">
                    <td class=\\"govuk-table__cell\\">
                        <div class=\\"govuk-form-group\\">
                            <div class=\\"govuk-checkboxes representation\\" data-module=\\"govuk-checkboxes\\">
                                <div class=\\"govuk-checkboxes__item\\">
                                    <input class=\\"govuk-checkboxes__input\\" id=\\"representation--1\\" name=\\"representationId\\"
                                    type=\\"checkbox\\" value=\\"1\\">
                                    <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"representation--1\\">Select representation 1</label>
                                </div>
                            </div>
                        </div>
                    </td>
                    <td class=\\"govuk-table__cell\\">mock-reference-1</td>
                    <td class=\\"govuk-table__cell\\">10 Oct 2023</td>
                    <td class=\\"govuk-table__cell\\"><strong class=\\"govuk-tag govuk-tag--\\">VALID</strong>
                    </td>
                </tr>
                <tr class=\\"govuk-table__row\\">
                    <td class=\\"govuk-table__cell\\">
                        <div class=\\"govuk-form-group\\">
                            <div class=\\"govuk-checkboxes representation\\" data-module=\\"govuk-checkboxes\\">
                                <div class=\\"govuk-checkboxes__item\\">
                                    <input class=\\"govuk-checkboxes__input\\" id=\\"representation--2\\" name=\\"representationId\\"
                                    type=\\"checkbox\\" value=\\"2\\">
                                    <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"representation--2\\">Select representation 2</label>
                                </div>
                            </div>
                        </div>
                    </td>
                    <td class=\\"govuk-table__cell\\">mock-reference-3</td>
                    <td class=\\"govuk-table__cell\\">10 Oct 2023</td>
                    <td class=\\"govuk-table__cell\\"><strong class=\\"govuk-tag govuk-tag--\\">VALID</strong>
                    </td>
                </tr>
                <tr class=\\"govuk-table__row\\">
                    <td class=\\"govuk-table__cell\\">
                        <div class=\\"govuk-form-group\\">
                            <div class=\\"govuk-checkboxes representation\\" data-module=\\"govuk-checkboxes\\">
                                <div class=\\"govuk-checkboxes__item\\">
                                    <input class=\\"govuk-checkboxes__input\\" id=\\"representation--3\\" name=\\"representationId\\"
                                    type=\\"checkbox\\" value=\\"3\\">
                                    <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"representation--3\\">Select representation 3</label>
                                </div>
                            </div>
                        </div>
                    </td>
                    <td class=\\"govuk-table__cell\\">mock-reference-3</td>
                    <td class=\\"govuk-table__cell\\">10 Oct 2023</td>
                    <td class=\\"govuk-table__cell\\"><strong class=\\"govuk-tag govuk-tag--\\">VALID</strong>
                    </td>
                </tr>
            </tbody>
        </table>
        <button type=\\"submit\\" class=\\"govuk-button\\" data-module=\\"govuk-button\\">Publish representations</button>
    </form>
    <script nonce=\\"EdcOUaJ8lczj9tIPO0lPow==\\">
        const selectAllRepsContainerElement = document.getElementById('select-all-representations-container');
            const selectAllRepsElement = document.getElementById('representationId');
            const repElements = document.querySelectorAll('.representation input');
            const numberOfSelectedRepsElement = document.getElementById('number-of-selected-representations');
        
            let numberOfSelectedReps;
        
            const updateNumberOfSelectedReps = () => {
              numberOfSelectedReps = 0;
              repElements.forEach((repElement) => {
                if (repElement.checked) numberOfSelectedReps++;
              });
              return numberOfSelectedReps;
            }
        
            const setNumberOfSelectedReps = () => numberOfSelectedRepsElement.innerHTML = numberOfSelectedReps;
        
            const updateAndSetNumberOfSelectedReps = () => {
              updateNumberOfSelectedReps()
              setNumberOfSelectedReps();
            }
        
            const isAllRepsSelected = () => numberOfSelectedReps === repElements.length;
        
            const toggleSelectAllReps = (check) => selectAllRepsElement.checked = check;
        
            selectAllRepsElement.addEventListener('change', function () {
              repElements.forEach((repElement) => repElement.checked = this.checked);
              updateAndSetNumberOfSelectedReps();
            })
        
            repElements.forEach((repElement) => {
              repElement.addEventListener('change', function () {
                updateAndSetNumberOfSelectedReps();
                if (!this.checked) toggleSelectAllReps(false);
                else if (isAllRepsSelected()) toggleSelectAllReps(true);
              })
            });
        
            selectAllRepsContainerElement.className = 'govuk-!-display-inline-block';
            updateAndSetNumberOfSelectedReps();
    </script>
</main>"
`;

exports[`publish-updated-representations.controller POST /applications-service/:caseId/relevant-representations/select-representations-for-publishing error should render the page with the error summary 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-two-thirds\\">
            <div class=\\"govuk-error-summary\\" data-module=\\"govuk-error-summary\\">
                <div role=\\"alert\\">
                    <h2 class=\\"govuk-error-summary__title\\"> There is a problem</h2>
                    <div class=\\"govuk-error-summary__body\\">
                        <ul class=\\"govuk-list govuk-error-summary__list\\">
                            <li><a href=\\"#representationId\\">You must select representations to publish</a>
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
        </div>
    </div><span class=\\"govuk-caption-m\\"> Publishing queue</span>
    <h2 class=\\"govuk-heading-l\\"> Select representations for publishing</h2>
    <p class=\\"govuk-body\\"><strong>Project name:</strong> mock title</p>
    <p class=\\"govuk-body\\">The queue contains <strong>3</strong> representations.</p>
    <br>
    <p class=\\"govuk-body\\">You must select the representations to publish</p>
    <p class=\\"govuk-body\\"><strong><span id=\\"number-of-selected-representations\\">0</span> representations on this page are selected.</strong>
    </p>
    <form method=\\"post\\">
        <table class=\\"govuk-table pins-files-list__table pins-table\\">
            <thead class=\\"govuk-table__head\\">
                <tr class=\\"govuk-table__row\\">
                    <th scope=\\"col\\" class=\\"govuk-table__header white-space--nowrap\\">
                        <div class=\\"govuk-!-display-none\\" id=\\"select-all-representations-container\\">
                            <div class=\\"govuk-form-group\\">
                                <div class=\\"govuk-checkboxes\\" data-module=\\"govuk-checkboxes\\">
                                    <div class=\\"govuk-checkboxes__item\\">
                                        <input class=\\"govuk-checkboxes__input\\" id=\\"representationId\\" name=\\"select-all-representations\\"
                                        type=\\"checkbox\\" value=\\"\\">
                                        <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"representationId\\">Select all representations</label>
                                    </div>
                                </div>
                            </div>
                        </div><span class=\\"govuk-!-display-inline-block\\">Select</span>
                    </th>
                    <th scope=\\"col\\" class=\\"govuk-table__header\\">Reference</th>
                    <th scope=\\"col\\" class=\\"govuk-table__header\\">From</th>
                    <th scope=\\"col\\" class=\\"govuk-table__header\\">Status</th>
                </tr>
            </thead>
            <tbody class=\\"govuk-table__body\\">
                <tr class=\\"govuk-table__row\\">
                    <td class=\\"govuk-table__cell\\">
                        <div class=\\"govuk-form-group\\">
                            <div class=\\"govuk-checkboxes representation\\" data-module=\\"govuk-checkboxes\\">
                                <div class=\\"govuk-checkboxes__item\\">
                                    <input class=\\"govuk-checkboxes__input\\" id=\\"representation--1\\" name=\\"representationId\\"
                                    type=\\"checkbox\\" value=\\"1\\">
                                    <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"representation--1\\">Select representation 1</label>
                                </div>
                            </div>
                        </div>
                    </td>
                    <td class=\\"govuk-table__cell\\">mock-reference-1</td>
                    <td class=\\"govuk-table__cell\\">10 Oct 2023</td>
                    <td class=\\"govuk-table__cell\\"><strong class=\\"govuk-tag govuk-tag--\\">VALID</strong>
                    </td>
                </tr>
                <tr class=\\"govuk-table__row\\">
                    <td class=\\"govuk-table__cell\\">
                        <div class=\\"govuk-form-group\\">
                            <div class=\\"govuk-checkboxes representation\\" data-module=\\"govuk-checkboxes\\">
                                <div class=\\"govuk-checkboxes__item\\">
                                    <input class=\\"govuk-checkboxes__input\\" id=\\"representation--2\\" name=\\"representationId\\"
                                    type=\\"checkbox\\" value=\\"2\\">
                                    <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"representation--2\\">Select representation 2</label>
                                </div>
                            </div>
                        </div>
                    </td>
                    <td class=\\"govuk-table__cell\\">mock-reference-3</td>
                    <td class=\\"govuk-table__cell\\">10 Oct 2023</td>
                    <td class=\\"govuk-table__cell\\"><strong class=\\"govuk-tag govuk-tag--\\">VALID</strong>
                    </td>
                </tr>
                <tr class=\\"govuk-table__row\\">
                    <td class=\\"govuk-table__cell\\">
                        <div class=\\"govuk-form-group\\">
                            <div class=\\"govuk-checkboxes representation\\" data-module=\\"govuk-checkboxes\\">
                                <div class=\\"govuk-checkboxes__item\\">
                                    <input class=\\"govuk-checkboxes__input\\" id=\\"representation--3\\" name=\\"representationId\\"
                                    type=\\"checkbox\\" value=\\"3\\">
                                    <label class=\\"govuk-label govuk-checkboxes__label\\" for=\\"representation--3\\">Select representation 3</label>
                                </div>
                            </div>
                        </div>
                    </td>
                    <td class=\\"govuk-table__cell\\">mock-reference-3</td>
                    <td class=\\"govuk-table__cell\\">10 Oct 2023</td>
                    <td class=\\"govuk-table__cell\\"><strong class=\\"govuk-tag govuk-tag--\\">VALID</strong>
                    </td>
                </tr>
            </tbody>
        </table>
        <button type=\\"submit\\" class=\\"govuk-button\\" data-module=\\"govuk-button\\">Publish representations</button>
    </form>
    <script nonce=\\"EdcOUaJ8lczj9tIPO0lPow==\\">
        const selectAllRepsContainerElement = document.getElementById('select-all-representations-container');
            const selectAllRepsElement = document.getElementById('representationId');
            const repElements = document.querySelectorAll('.representation input');
            const numberOfSelectedRepsElement = document.getElementById('number-of-selected-representations');
        
            let numberOfSelectedReps;
        
            const updateNumberOfSelectedReps = () => {
              numberOfSelectedReps = 0;
              repElements.forEach((repElement) => {
                if (repElement.checked) numberOfSelectedReps++;
              });
              return numberOfSelectedReps;
            }
        
            const setNumberOfSelectedReps = () => numberOfSelectedRepsElement.innerHTML = numberOfSelectedReps;
        
            const updateAndSetNumberOfSelectedReps = () => {
              updateNumberOfSelectedReps()
              setNumberOfSelectedReps();
            }
        
            const isAllRepsSelected = () => numberOfSelectedReps === repElements.length;
        
            const toggleSelectAllReps = (check) => selectAllRepsElement.checked = check;
        
            selectAllRepsElement.addEventListener('change', function () {
              repElements.forEach((repElement) => repElement.checked = this.checked);
              updateAndSetNumberOfSelectedReps();
            })
        
            repElements.forEach((repElement) => {
              repElement.addEventListener('change', function () {
                updateAndSetNumberOfSelectedReps();
                if (!this.checked) toggleSelectAllReps(false);
                else if (isAllRepsSelected()) toggleSelectAllReps(true);
              })
            });
        
            selectAllRepsContainerElement.className = 'govuk-!-display-inline-block';
            updateAndSetNumberOfSelectedReps();
    </script>
</main>"
`;
